src/auth/OAUthWithIDTokenValidation.js

```diff
@@ -77,11 +83,15 @@ OAUthWithIDTokenValidation.prototype.create = function(params, data, cb) {
             audience: this.clientId,
             issuer: 'https://' + this.domain + '/'
           },
-          function(err, payload) {
-            if (err) {
-              return rej(err);
+          function(err) {
+            if (!err) {
+              return res(r);
+            }
+            if (err.message && err.message.includes(HS256_IGNORE_VALIDATION_MESSAGE)) {
+              console.warn(HS256_IGNORE_VALIDATION_MESSAGE);
```

Comment:
I'd move this `console.warn` into the `if (!_this.clientSecret) {` above. Since the "error" you're passing on the callback is going to get thrown away in this scenario, you could add a custom property easier to check such as a flag instead of a string/message.

e.g. above

```js
         if (!_this.clientSecret) {
            console.warn(HS256...);
            return callback({ skip_id_token_validation: true });
          }
```

then here

```js
       function(err) {
            if (!err || err.skip_id_token_validation) {
              return res(r);
            }
            return rej(err);
      }
```

Reply:
Nice to know that we think the same 🎉 but, sadly, this doesn't work (it was my first attempt): https://github.com/auth0/node-auth0/pull/330/commits/5aa546445a81d52af832b899732faf597f290ec7

Basically, jsonwebtoken doesn't support custom errors, it will accept only the message (https://github.com/auth0/node-jsonwebtoken/blob/master/verify.js#L86).